### PR TITLE
kubeadm: APIEndpointFromString verifies ip

### PR DESCRIPTION
**Pull request by chymy**
Signed-off-by: chymy <chang.min1@zte.com.cn>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
`Apiendpointfromstring` should verify the validity of IP.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/apis/kubeadm/apiendpoint.go [hunk 1]
```diff
@@ -29,6 +29,9 @@ func APIEndpointFromString(apiEndpoint string) (APIEndpoint, error) {
 	if err != nil {
 		return APIEndpoint{}, errors.Wrapf(err, "invalid advertise address endpoint: %s", apiEndpoint)
 	}
+	if net.ParseIP(apiEndpointHost) == nil {
+		return APIEndpoint{}, errors.Wrapf(errors.New("invalid IP"), "invalid advertise address endpoint IP: %s", apiEndpointHost)
```
Comment (neolit123) [inline cmd/kubeadm/app/apis/kubeadm/apiendpoint.go:33]:
```suggestion
		return APIEndpoint{}, errors.Errorf("invalid API endpoint IP: %s", apiEndpointHost)
```

please keep the commits squashed to one after this change.

**Review by neolit123 — COMMENTED** [review]
this seems to agree with the source code and docs:
https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/https://github.com/kubernetes/kubernetes/blob/cea1d4e20b4a7886d8ff65f34c6d4f95efcb4742/staging/src/k8s.io/apiserver/pkg/server/options/serving.go#L39

/triage accepted
/priority backlog

**Comment by fabriziopandini** [comment]
/ok-to-test
SGTM to me

File: cmd/kubeadm/app/apis/kubeadm/apiendpoint.go (hunk 1 above)
Comment (chymy) [reply cmd/kubeadm/app/apis/kubeadm/apiendpoint.go]: Thanks @neolit123 , fixed.

**Review by neolit123 — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**